### PR TITLE
refactor(core-backend): rename generics `core-backend` & `core-processor` so they emphasize their essence

### LINKED ISSUE #2654: Rename all generics in the `core-backend` so they emphasize their nature/purposes
### Proposal

For example, `EP` can be changed to `EntryPoint`, `PrepMem` to `MemoryPreparationError`, `T` in `Environment::execute` to `MemoryPreparationError`.

It's hard to navigate in the code with short unclear generics names.

### PR DESCRIPTION
**Pull request by SabaunT**
Resolves #2654 

This PR actually **renames** only several traits/types:
- `EnvironmentExecutionError` -> `EnvironmentError` (as environment actually doesn't execute, errors aren't much related to execution).
- `ProcessorError` -> `ExtError` (`ProcessorExt` trait can have `ProcessorExt` error, but there's no such associated type in this trait. What's more there's no additional helping context with "Processor*" prefix to the error, especially if we look at the type definition). Same for `ProcessorAllocError` -> `ExtAllocError`.

All other changes are **name extensions** (instead of short names or one-letter generic names, which made it hard to understand the code and definitions of some types):
- generic `EP` -> EntryPoint
- generic `A` in some functions of `core_processor::execute` -> `ProcessorExt`
- trait `Ext` -> `Externalities`
- trait `ProcessorExt` -> `ProcessorExternalities`
- generic `PrepMem` (and just `T` in some places) -> `PrepareMemoryError`
- generic `E` -> `Ext`. That was the most important as `E` is used in the same crate/package for `Environment`, some error and `Externalities` too.

Some rules used for changes:
- Traits are given full names
- Generics, associated types and common types can have either full or short name (like `Externalities` trait implementor is `Ext`).

### CODE REVIEW

**Review by gshep — APPROVED** [review]
👍

File: core-backend/common/src/lib.rs [hunk 1]
```diff
@@ -246,83 +247,99 @@ pub trait BackendExt: EnvExt + CountersOwner {
...
-        entry_point: EP,
+        entry_point: EntryPoint,
         entries: BTreeSet<DispatchKind>,
         mem_size: WasmPage,
-    ) -> Result<Self, EnvironmentExecutionError<Self::Error, Infallible>>;
+    ) -> Result<Self, EnvironmentError<Self::SystemError, Infallible>>;
 
     /// Run instance setup starting at `entry_point` - wasm export function name.
-    fn execute<F, T>(self, pre_execution_handler: F) -> EnvironmentExecutionResult<T, Self, EP>
+    fn execute<PrepareMemory, PrepareMemoryError>(
```
Comment (shamilsan) [inline core-backend/common/src/lib.rs:332]: As for me, we could leave the `F` and `E` generics here as well-known and readable within the context.
```suggestion
    fn execute<F, E>(
```

File: core-backend/common/src/lib.rs [hunk 2]
```diff
@@ -356,10 +373,10 @@ pub trait BackendState {
     }
 
     /// Process alloc function result
-    fn process_alloc_func_result<T: Sized, E: BackendAllocExtError>(
+    fn process_alloc_func_result<T: Sized, ExtAllocError: BackendAllocExternalitiesError>(
```
Comment (shamilsan) [inline core-backend/common/src/lib.rs:376]: <T, E> is also quite a stable pattern for generics returning `Result`.

File: core-backend/sandbox/src/env.rs [hunk 3]
```diff
@@ -339,12 +342,19 @@ where
         }
     }
 
-    fn execute<F, T>(self, pre_execution_handler: F) -> EnvironmentExecutionResult<T, Self, EP>
+    fn execute<PrepareMemory, PrepareMemoryError>(
```
Comment (shamilsan) [inline core-backend/sandbox/src/env.rs:345]: Also, I'd leave `<F, E>`:
```suggestion
    fn execute<F, E>(
```

File: core-backend/sandbox/src/env.rs [hunk 4]
```diff
@@ -384,7 +394,7 @@ where
             unreachable!("We cannot use sandbox backend in std environment currently");
         };
 
-        match pre_execution_handler(&mut runtime.memory, stack_end, globals_config) {
+        match prepare_memory(&mut runtime.memory, stack_end, globals_config) {
```
Comment (shamilsan) [inline core-backend/sandbox/src/env.rs:397]: What if we add some additional functionality to this function?

**Review by shamilsan — APPROVED** [review]
👍

File: core-backend/sandbox/src/env.rs (hunk 4 above)
Comment (SabaunT) [reply core-backend/sandbox/src/env.rs:397]: Currently it's a handler, which returns `PrepareMemoryError`, so renaming that seems fine. If complex functionality is introduced, not only name, but the closure signature will change.

File: core-backend/common/src/lib.rs (hunk 1 above)
Comment (SabaunT) [reply core-backend/common/src/lib.rs:332]: I can leave `F`, but I'd like to move from `E` as it was used for errors, environment, ext. what do you think?

File: core-backend/common/src/lib.rs (hunk 2 above)
Comment (SabaunT) [reply core-backend/common/src/lib.rs:376]: In this module we have another method, which accepts `Result` with a generic error type. Although `<T, E>` is a suitable pattern, I'd move from short generics in this crate/package as it lowers readability.

File: core-backend/common/src/lib.rs (hunk 1 above)
Comment (SabaunT) [reply core-backend/common/src/lib.rs:332]: agreed to resolve

File: core-backend/common/src/lib.rs (hunk 2 above)
Comment (SabaunT) [reply core-backend/common/src/lib.rs:376]: agreed to resolve

File: core-backend/sandbox/src/env.rs (hunk 4 above)
Comment (SabaunT) [reply core-backend/sandbox/src/env.rs:397]: agreed to resolve

**Pull request merged**